Updated the polyfill to WebVR 1.1

toji opened the pull request:
Also includes functionality to shim native 1.0 implementations to the 1.1 spec.

brianchirls commented:
The logic around `WebVRSpecShim` is a little weird and could be improved IMHO.

As currently written [here](https://github.com/borismus/webvr-polyfill/pull/149/commits/51aab8dbe4447ddf523da484180ddc19e611563b#diff-7bfb91b61bb80cff41aba297525c4dc7R38), the frameData stuff will be shimmed automatically if there's a native WebVR implementation. But if you need the polyfill, then you need to run `InitializeSpecShim`. So if you forget to run `InitializeSpecShim`, then you're stuck with a broken polyfill.

Also, there's no need for `WebVRSpecShim` to be a constructor, since it has neither methods nor properties. (On that note, I'm not sure the polyfill needs to be one either, but one thing at a time.)

So I suggest automatically polyfilling `VRFrameData` et al right inside `WebVRPolyfill` every time. There's no cost to doing so, since you're loading the JS code anyway, and it won't have any effect if it's not needed.

borismus commented:
This is a large change and we don't have unit tests :( 

Could you test on iOS, Android, and various desktop browsers?

toji commented:
Fixed the desktop issues and a few more besides, incorporated @brianchirls feedback too. Tested on Android, iOS, and Desktop Chrome/Firefox/Safari (It's installed on https://webvr.info/samples if you want to test yourself.)

cvan commented on src/main.js:
```diff
@@ -72,4 +73,10 @@ if (!window.WebVRConfig.DEFER_INITIALIZATION) {
+  // Call this if you want to use the shim without the rest of the polyfill.
+  // InitializeWebVRPolyfill() will install the shim automatically when needed,
+  // so this should rarely be used.
+  window.InitializeSpecShim = function() {
```
IMO, we should not be adding yet another global for the WebVR Polyfill. see #47; I have a branch of this I've done locally that moves everything to a single `WEBVR_POLYFILL`  global that is exported from `./webvr-polyfll.js` and then from here in `src/main.js`.

borismus replied:
Agree. When would this global be called? Also why do we have the two different names `InstallWebVRSpecShim` and `InitializeSpecShim` ?
